### PR TITLE
Feature/farms search

### PR DESCRIPTION
**Pull request by memacius32**
- Adding Search feature for pools page, Filtering by token or network name, or address. 
- Adding pools sorting by apr or liquidity.
- Adding render optimizations.

### CODE REVIEW

File: src/utils/pools.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  //      If TRI is the other token, it'll be first
+  if (currency0 === CETH || currency1 === CETH) {
+    return currency0 === CETH ? token1IsFirst : token0IsFirst
+  }
+
+  // If pair has TRI, put TRI first
+  return token0.equals(TRI[token0.chainId]) ? token0IsFirst : token1IsFirst
+}
+
+export const poolIsStaking = (stakedAmount: TokenAmount | null) => {
```
Comment (specialist0x) [inline src/utils/pools.ts:32]: This shouldn't be called `poolIsStaking` bc that implies it has context of the user
```suggestion
export const isTokenAmountPositive = (stakedAmount: TokenAmount | null) => {
```

File: src/pages/EarnTri/EarnTri.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+export const StyledToggleContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+   margin-bottom: 1rem;
+  `};
+`
+
+export const StyledSortContainer = styled.div``
```
Comment (specialist0x) [inline src/pages/EarnTri/EarnTri.styles.tsx:81]: Is this needed?

File: src/pages/EarnTri/index.tsx [hunk 3]
```diff
@@ -1,143 +1,198 @@
...
+        return sortDescending
+          ? farmArrs.sort((a, b) => (a.totalStakedInUSD < b.totalStakedInUSD ? 1 : -1))
+          : farmArrs.sort((a, b) => (a.totalStakedInUSD > b.totalStakedInUSD ? 1 : -1))
+    }
+    switch (sortBy) {
+      case SortingType.totalApr:
+        return sortDescending
+          ? farmArrs.sort((a, b) => (a.apr + a.apr2 < b.apr + b.apr2 ? 1 : -1))
+          : farmArrs.sort((a, b) => (a.apr + a.apr2 > b.apr + b.apr2 ? 1 : -1))
+    }
```
Comment (specialist0x) [inline src/pages/EarnTri/index.tsx:75]: Was there a reason for multiple `switch`s?

```suggestion
    switch (sortBy) {
      case SortingType.default:
        return POOLS_ORDER.map(index => allFarmArrs[index])
      case SortingType.liquidity:
        return sortDescending
          ? farmArrs.sort((a, b) => (a.totalStakedInUSD < b.totalStakedInUSD ? 1 : -1))
          : farmArrs.sort((a, b) => (a.totalStakedInUSD > b.totalStakedInUSD ? 1 : -1))
      case SortingType.totalApr:
        return sortDescending
          ? farmArrs.sort((a, b) => (a.apr + a.apr2 < b.apr + b.apr2 ? 1 : -1))
          : farmArrs.sort((a, b) => (a.apr + a.apr2 > b.apr + b.apr2 ? 1 : -1))
    }
```

**Comment by specialist0x** [comment]
Works great!

Couple things:

1. Looks like Legacy Farms still shows, even if I'm not deposited:

image

2. on `EarnTri/index.tsx`, can you also add support for LP addresses?  Should be similar to your existing implementation with the addition of `allFarmArrs.map(v => v.lpAddress)`

3. I pushed up https://github.com/trisolaris-labs/interface/pull/128, make sure to handle the conflict or rebase ahead of it

File: src/pages/EarnTri/EarnTri.styles.tsx (hunk 2 above)
Comment (memacius32) [reply src/pages/EarnTri/EarnTri.styles.tsx:81]: nope, leftover, cleaning 🧹

File: src/pages/EarnTri/index.tsx (hunk 3 above)
Comment (memacius32) [reply src/pages/EarnTri/index.tsx:75]: messed up!

**Comment by memacius32** [comment]
Rebased and fixed. 

Also I was wondering, do you agree it would make sense to filter addresses only with by exact match? I don't imagine anyone manually typing addresses
 Also our Swap and Farm search inputs say "...or paste address" in the placeholder

**Comment by specialist0x** [comment]
> Also I was wondering, do you agree it would make sense to filter addresses only with by exact match?

Well, partial matches might be used? They'd also be able to see the card to visually verify

> Also our Swap and Farm search inputs say "...or paste address" in the placeholder
I think `Search farms` or `Search` works

**Comment by memacius32** [comment]
Alright, I leave the current functionality then, it filters with partial matches.

About the placeholder, what about "Search farms by name or address" ?🤔

**Comment by specialist0x** [comment]
> About the placeholder, what about "Search farms by name or address" ?

Let's do `Search farms by token or address`

**Comment by memacius32** [comment]
@specialist0x done

**Review by specialist0x — APPROVED** [review]

**Pull request merged**